sklearn/naive_bayes.py

```diff
@@ -1420,8 +1426,11 @@ def _update_feature_log_prob(self, alpha):
     def _joint_log_likelihood(self, X):
         self._check_n_features(X, reset=False)
         jll = np.zeros((X.shape[0], self.class_count_.shape[0]))
+        is_sparse = sp.issparse(X)
         for i in range(self.n_features_in_):
             indices = X[:, i]
+            if is_sparse:
+                indices = indices.toarray().flatten()
```

Comment:
Same here regarding CSC and not needing to denstify in that case.

Reply:
I may be wrong, but even though the matrix is CSC, `X[:, i]` is a sparse matrix and cannot it used as indexing?

Reply:
I understand it now. You suggested I use indices and indptr to access to the column directly.

> csc_matrix((data, indices, indptr), [shape=(M, N)])
is the standard CSC representation where the row indices for column i are stored in indices[indptr[i]:indptr[i+1]] and their corresponding values are stored in data[indptr[i]:indptr[i+1]]. If the shape parameter is not supplied, the matrix dimensions are inferred from the index arrays.